src/index.ts

```diff
@@ -0,0 +1,224 @@
+ * @throws Error if the decoded codec did not matched Swarm Feed codec
+ */
+export function decodeFeedCid (cid: CID | string): Reference {
+  const result = _decodeReference(cid)
+
+  if (result.type !== ReferenceType.FEED) {
+    throw new Error('CID did not have Swarm Feed codec!')
+  }
+
+  return result.reference
+}
+
+/**
+ * Function to decode Manifest CID (both from string or CID instance) into hex encoded Swarm reference.
+ *
+ * @param cid
+ * @throws Error if the decoded codec did not matched Swarm Manifest codec
+ */
+export function decodeManifestCid (cid: CID | string): Reference {
+  const result = _decodeReference(cid)
+
+  if (result.type !== ReferenceType.MANIFEST) {
+    throw new Error('CID did not have Swarm Manifest codec!')
+  }
+
+  return result.reference
+}
+
+/**
+ * Decode CID or base encoded CID string into DecodeResult interface.
+ * Does not throw exception if the codec was not Swarm related. In that case `type` is undefined.
+ *
+ * @see DecodeResult
+ * @param cid
+ */
+export function decodeCid (cid: CID | string): DecodeResult {
+  return _decodeReference(cid)
+}
+
+function _decodeReference (cid: CID | string): DecodeResult {
+  if (typeof cid === 'string') {
+    cid = CID.parse(cid)
+  }
+
+  return {
+    reference: bytesToHex(cid.multihash.digest),
+    type: CODEC_TYPE_MAPPING[cid.code]
+  }
+}
+
+function _encodeReference (ref: string | Reference, codec: number): CID {
+  assertReference(ref)
+
+  const hashBytes = hexToBytes(ref)
+  return CID.createV1(codec, createMultihashDigest(KECCAK_256_CODEC, hashBytes))
+}
+
+type FlavoredType<Type, Name> = Type & { __tag__?: Name }
+
+/**
+ * Nominal type to represent hex strings WITHOUT '0x' prefix.
+ * For example for 32 bytes hex representation you have to use 64 length.
+ */
+type HexString<Length extends number = number> = FlavoredType<string & {
+  readonly length: Length
+}, 'HexString'>
+
+interface Bytes<Length extends number> extends Uint8Array {
+  readonly length: Length
+}
+
+/**
+ * Converts a hex string to Uint8Array
+ *
+ * @param hex string input without 0x prefix!
+ */
+function hexToBytes<Length extends number, LengthHex extends number = number> (
+  hex: HexString<LengthHex>
+): Bytes<Length> {
+  assertHexString(hex)
+
+  const bytes = new Uint8Array(hex.length / 2)
+  for (let i = 0; i < bytes.length; i++) {
+    const hexByte = hex.substr(i * 2, 2)
+    bytes[i] = parseInt(hexByte, 16)
+  }
+
+  return bytes as Bytes<Length>
+}
+
+/**
+ * Converts array of number or Uint8Array to HexString without prefix.
+ *
+ * @param bytes   The input array
+ * @param len     The length of the non prefixed HexString
+ */
+function bytesToHex<Length extends number = number> (bytes: Uint8Array, len?: Length): HexString<Length> {
+  const hexByte = (n: number) => n.toString(16).padStart(2, '0')
+  const hex = Array.from(bytes, hexByte).join('') as HexString<Length>
+
+  // TODO: Make Length mandatory: https://github.com/ethersphere/bee-js/issues/208
```

Comment:
are these relevant here?

Reply:
I guess not. This should be taken from some utils package that we gonna make once we get to the modularization. I just did not want to add whole bee-js as dependencies for these.